AdyenDropIn/Components/PaymentMethodListComponent.swift

```diff
@@ -97,13 +95,14 @@ internal final class PaymentMethodListComponent: ComponentLoader, PresentableCom
     ///
     /// - Parameter component: The component for which to start a loading animation.
     internal func startLoading(for component: PaymentComponent) {
-        let allListItems = listViewController.sections.flatMap(\.items)
-        let allComponents = componentSections.map(\.components).flatMap { $0 }
+        let allPaymentMethods = paymentMethodSections.map(\.paymentMethods).flatMap { $0 }
         
-        guard let index = allComponents.firstIndex(where: { $0 === component }) else {
+        guard let index = allPaymentMethods.firstIndex(where: { $0.type == component.paymentMethod.type }) else {
```

Comment:
maybe there are multiple payment methods with the same type, like for example gift card PM's I think have the same type `giftcard`, but not sure though, can you check that, or try to find something else to identify the payment method, like adding a UUID to the payment method, generated in the initializer?

Reply:
how about adding a function to compare payment methods?
```swift
    func isEqual(to other: PaymentMethod) -> Bool {
        switch (self as? StoredPaymentMethod, other as? StoredPaymentMethod) {
        case (.some(let lhs), .some(let rhs)):
            return lhs.identifier == rhs.identifier
        case (.none, .none):
            return self.name == other.name && self.type == other.type
        case (_, _):
            return false
        }
    }
```